### ISSUE TITLE
runrestic hangs when restic unexpectedly exits

### ISSUE THREAD

**Issue by viq** [sinnwerkstatt/runrestic#41]
```
# cat /etc/runrestic/home.toml
name = "home directories"
repositories = [ "rclone:test",]

[execution]
parallel = true
retry_count = 10
retry_backoff = "1:00 exponential"

[environment]
RESTIC_PASSWORD_FILE = "/etc/secrets/restic.password"
RCLONE_CONFIG = "/etc/secrets/rclone.conf"

[backup]
sources = [ "/home",]
exclude_patterns = [ "/home/_sysupgrade",]

[prune]
keep-last = 2
group-by = "host,paths"

[check]
checks = [ "check-unused", "read-data",]

[metrics.prometheus]
path = "/tmp/runrestic_home.prom"
```
```
# time /usr/local/virtualenvs/runrestic/bin/runrestic -c /etc/runrestic/home.toml -l debug -- --tag home
Parsing configuration file: /etc/runrestic/home.toml
[Environment] RESTIC_PASSWORD_FILE=/etc/secrets/restic.password
[Environment] RCLONE_CONFIG=/etc/secrets/rclone.conf
Spawning "['restic', '-r', 'rclone:test', 'backup', '--tag', 'home', '--exclude', '/home/_sysupgrade', '/home']"
Spawning "['restic', '-r', 'rclone:test', 'forget', '--tag', 'home', '--keep-last', '2', '--group-by', 'host,paths']"
Spawning "['restic', '-r', 'rclone:test', 'prune', '--tag', 'home']"
```
and just hangs there until I `^C` out of it. Meanwhile
```
# RESTIC_PASSWORD_FILE=/etc/secrets/restic.password RCLONE_CONFIG=/etc/secrets/rclone.conf restic -r rclone:test prune --tag home ; echo $?
unknown flag: --tag
1
```

**Comment by viq** [comment]
And while experimenting with things, I managed to get my repository dirty. Which caused `restic check` return with fatal error. Which was again not reported by runrestic, and it was just hanging there, waiting for something useful to happen.

**Comment by palto42** [comment]
The PR #52 should have fixed this issue.

**Comment by palto42** [comment]
Closing this, assuming it was fixed by #52
